contracts/stake-cw721/src/contract.rs

```diff
@@ -0,0 +1,1844 @@
+                    .add_submessages(hook_msgs)
+                    .add_attribute("action", "unstake")
+                    .add_attribute("from", info.sender)
+                    .add_attribute("token_id", token_id)
+                    .add_attribute("claim_duration", "None"));
+            }
+
+            let cw_send_msg = cw20::Cw20ExecuteMsg::Transfer {
+                recipient: validated_reward_wallet
+                    .clone()
+                    .unwrap_or_else(|| DEFAULT_ADDR.clone())
+                    .to_string(),
+                amount: amount_to_claim,
+            };
+
+            let wasm_claim_rewards_msg = cosmwasm_std::WasmMsg::Execute {
+                contract_addr: config.reward_token_address.unwrap().to_string(),
+                msg: to_binary(&cw_send_msg)?,
+                funds: vec![],
+            };
+
+            Ok(response
+                .add_message(wasm_unstake_nft_msg)
+                .add_submessages(hook_msgs)
+                .add_attribute("action", "unstake")
+                .add_attribute("from", info.sender)
+                .add_attribute("token_id", token_id)
+                .add_attribute("claim_duration", "None")
+                .add_message(wasm_claim_rewards_msg)
+                .add_attribute(
+                    "reward_wallet",
+                    validated_reward_wallet.unwrap_or_else(|| DEFAULT_ADDR.clone()),
+                )
+                .add_attribute("reward_amount", amount_to_claim))
+        }
+
+        Some(duration) => {
+            let outstanding_claims = NFT_CLAIMS
+                .query_claims(deps.as_ref(), &info.sender)?
+                .nft_claims;
+            if outstanding_claims.len() >= MAX_CLAIMS as usize {
+                return Err(ContractError::TooManyClaims {});
+            }
+
+            NFT_CLAIMS.create_nft_claim(
+                deps.storage,
+                &info.sender,
+                token_id.clone(),
+                duration.after(&env.block),
+            )?;
+
+            if amount_to_claim.is_zero() || validated_reward_wallet == None {
+                return Ok(Response::new()
+                    .add_attribute("action", "unstake")
+                    .add_submessages(hook_msgs)
+                    .add_attribute("from", info.sender)
+                    .add_attribute("token_id", token_id)
+                    .add_attribute("claim_duration", format!("{}", duration)));
+            }
+
+            let outstanding_reward_claims = REWARD_CLAIMS
+                .query_claims(deps.as_ref(), &info.sender)?
+                .claims;
+            if outstanding_reward_claims.len() >= MAX_CLAIMS as usize {
+                return Err(ContractError::TooManyClaims {});
+            }
+
+            REWARD_CLAIMS.create_claim(
+                deps.storage,
+                &validated_reward_wallet
+                    .clone()
+                    .unwrap_or_else(|| DEFAULT_ADDR.clone()),
```

Comment:
Seems like we just shouldn't create the claim if the reward wallet is `None`? Sending to an invalid address seems liable to cause confusion.

Reply:
Done.